pkg/containerimagebuilderpusher/kaniko.go

```diff
@@ -637,6 +637,20 @@ func (k *Kaniko) resolveFailFast(ctx context.Context,
 				continue
 			}
 			if jobPod.Status.Phase == v1.PodPending || jobPod.Status.Phase == v1.PodUnknown {
+				if failure, failed := k.getLastPodWarningEvent(ctx, namespace, jobPod.Name); failed {
+					errorMessage := fmt.Sprintf("%s event for Kaniko pod %s. Message: %s",
+						failure.Reason,
+						jobPod.Name,
+						failure.Message)
+
+					// if an error has changed, print it to the logs
+					if errorMessage != lastError {
+						buildLogger.WarnWithCtx(ctx, fmt.Sprintf("%s event for Kaniko pod", failure.Reason),
```

Comment:
It still reads a bit weird, and its best the the message itself will not contain a variable and be consistent so we can debug it easier in the future. I prefer this:
```suggestion
						buildLogger.WarnWithCtx(ctx, 
						        "Kaniko pod received a warning event", 
						        "eventReason", failure.Reason,
							"eventMessage", failure.Message,
							"podName", jobPod.Name)
```